server.js

```diff
@@ -7,23 +7,116 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+//___________________Model
+const Thought = mongoose.model('Thought', {
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  name: {
+    type: String,
+    default: "Anonymous"
+  },
+  createdAt: {
+    type: Date,
+    default: () => new Date()
+  }
+})
+
+//___________________Defines port app till run on
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+//___________________Middleweares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
+const listEndpoints = require('express-list-endpoints')
+
+//___________________Error messages
+const SERVICE_UNAVAILABLE = "service unavailable"
+const BAD_REQUEST = "Bad request"
+const POST_THOUGHT_FAILED = "Could not save thought"
+const ID_NOT_FOUND = "Thought ID was not found"
+
+//___________________Error messages if server is not connected
+app.use((req, res, next) => {
+  if(mongoose.connection.readyState === 1) {
+    next();
+  } else {
+    res
+    .status(503)
+    .send({ error: SERVICE_UNAVAILABLE});
+  };
+});
 
-// Start defining your routes here
+//___________________Defining routes
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+//___________________GET all thoughts
+app.get('/thoughts', async (req, res) => {
+  const { sort, page } = req.query
+
+  const pageNumber = +page || 1; 
+  const pageSize = 8 * pageNumber;
+  
+  const totalThoughts = await Thought.find();
+  const numberOfThoughts = totalThoughts.length
+  
+  const sortThoughts = (sort) => {
+    if (sort === "likes") {
+      return { hearts: -1 }  
+    }
+    else if (sort === "oldest") {
+      return { createdAt: 1 }
+    }
+    else {
+     return { createdAt: - 1 }
+    }
+  } 
+
+  const thoughts = await Thought.find()
+    .sort(sortThoughts(sort))
+    .limit(pageSize)
+    .exec()
+
+    if (thoughts) {
+    res.status(200).send({total: numberOfThoughts, results: thoughts})
+  } else {
+    res.status(400).send({ error: BAD_REQUEST, error: err.errors })
+  }
+})
+
+//___________________POST thought
+app.post('/thoughts', async (req, res) => {
+  const { name, message } = req.body
+  try {
+    const thought = await new Thought({ message: message, name: name}).save()
```

Comment:
This could probably be written as just `{message, name}` since the fields are named the same